### ISSUE TITLE
Coming Soon v2: Site hasn't been launched, but it is Public at the same time

### ISSUE THREAD

**Issue by ivan-ottinger** [Automattic/wp-calypso#48316]
The site hasn't been launched and at the same time it is public.

![Markup on 2020-12-14 at 16:35:02](https://user-images.githubusercontent.com/25105483/102101009-66d95600-3e2a-11eb-8bf5-3676f777a962.png)

It is not clear how this happened and how to reproduce the issue, but I wanted to make sure we have a record in case the issue occurs again.

Site ID: 186597945 (I am not sharing the site URL on purpose as this GH issue is public)
Ticket with more details: 3570337-zd-woothemes

#### What I expected

The site shouldn't be visible to public when it hasn't been launched yet.

#### What happened instead

The site is public.

#### Context / Source

#user-report, 3570337-zd-woothemes

**Comment by TeniCola** [comment]
I've noticed a few cases where the issue is the direct inverse. For my users, they launched their site and set it to public, but the Coming Soon banner + prompt on the live site _persists_. No matter how many times the settings were saved, it wouldn't actually make it live, and the user reports their visitors are unable to see their site.

Cases are in 3628418-zen and 3611359-zen with a link to the current discussion and video

The current workaround for my users were to temporarily switch the sites to Coming Soon, and then switching them back to Public (saving Site Settings in between).

I am not sure if a similar workaround exists for this initial report; I imagine we can let the user launch their site, and then set it back to Coming Soon, until this issue can be replicated and solved

**Comment by ivan-ottinger** [comment]
Interesting. I have actually encountered a similar issue yesterday: 26838449-hc

I thought that it was just some browser glitch, but when we have several reports already, it will be something else.

**Comment by ivan-ottinger** [comment]
Another case that seems related: 26864179-hc. The user reported that their images are not loading. The site seemed public, but the "Private" label was visible in Calypso.

Once I switched the site to "Coming soon" and then back to "Public" images started loading correctly.

**Comment by Nic-Sevic** [comment]
Another case with site marked as not yet launched but is visible. In this case it looks like the settings in WP Admin were changed to allow search engine indexing but this wasn't reflected on the Calypso side.

Site is visible when logged in to WP and when not, can also pulled with API calls.

![](https://d.pr/i/u514eG+)
Link to image: https://d.pr/i/u514eG

![](https://d.pr/i/HUHknX+)
Link to image: https://d.pr/i/HUHknX

3624752-zen

**Comment by ramonjd** [comment]
Hi all!

Is it happening on newly-created sites? And are they only Atomic? Thanks!

**Comment by ivan-ottinger** [comment]
> Is it happening on newly-created sites? And are they only Atomic? Thanks!

Here are the details on the issues I reported above:

- 3570337-zd-woothemes - Personal / Simple; the site was created on 2020-12-08; the issue noticed on 2020-12-14
- 26838449-hc - Premium / Simple; the site was created on 2021-01-07; the issue noticed on the same day when the user wanted to make the site Public
- 26864179-hc - Premium / Simple; the site was created on 2020-04-12; the issue noticed on 2021-01-08 when the user tried to make the site Public; If that helps, this site was previously switched from Private to Public on 2021-04-12 and then from Public to Private on 2020-10-21.

**Comment by ramonjd** [comment]
Hmm, I wonder if there's some delay after we update the `launch-status` option.

Or it could be that Calypso store needs a little pat on the shoulder to update its own records.

@roo2 do you have time to debug this today? Thanks!

**Comment by TeniCola** [comment]
Another case in 27088451-hc -- Site first created 3 weeks ago on 2020-12-29 and apparently also launched around that same time, but the issue was noticed today.

I had to toggle privacy in the Settings until the "Coming Soon" marker no longer appeared in the sidebar to help the user, as clearing their cache and other fixes did not work in their previous chats.

**Comment by ramonjd** [comment]
> I had to toggle privacy in the Settings until the "Coming Soon" marker no longer appeared in the sidebar to help the user,

I wonder if it's related to what I was trying to fix over at: https://github.com/Automattic/wp-calypso/pull/48153

@roo2 any chance you could take a look to see if the redux store is being update correctly after launch? Cheers

**Comment by roo2** [comment]
I found that we could get into the "public, unlaunched" state pretty easily by setting the site to public from wp-admin, this was updating wpcom_public_coming soon but not launch-status, oops! D55837-code should fix it. The issues that @ivan-ottinger found seem a bit different, I'll keep looking into them 🤔

**Comment by kylemcph** [comment]
I've seen a number of cases recently of sites set to public, but still in "Coming Soon" mode. Screenshot to show what I mean: 

![](https://d.pr/i/Q9Yjka+)
Direct link to image: https://d.pr/i/Q9Yjka

I haven't been able to replicate on my end. I asked the user from 27324642-hc what steps they took to make site public and they couldn't remember. The site in question was a simple site. 

Edit to add that this is resolved by making the site private, then public again.

**Comment by ramonjd** [comment]
> I've seen a number of cases recently of sites set to public, but still in "Coming Soon" mode. Screenshot to show what I mean:

Thanks @kylemcph

Hoping D55837-code will prevent this in the future. I think @roo2 has it nearly ready to ship.

**Comment by thehenrybyrd** [comment]
Ran into another of these in #27451479-hc

**Comment by Nic-Sevic** [comment]
another here; fixed with coming soon > public toggle
27380112-hc

**Comment by Nic-Sevic** [comment]
another public but not launched: 
3720706-zen

**Comment by ramonjd** [comment]
D55837-code has been shipped. If you see further instances of this, please let us know. Along with steps to reproduce. 🤞 
Thank you!

**Comment by ramonjd** [comment]
I've come across this a couple of times today again. 

D55837-code did fix an edge case, which needed to be done, but I'm seeing it immediately after having created a site (only sometimes).

<img width="726" alt="Screen Shot 2021-02-11 at 12 23 36 pm" src="https://user-images.githubusercontent.com/6458278/107603404-c0220200-6c80-11eb-8a13-fdaa812c321c.png">


 It appears intermittent. For some reason it appears we're not correctly setting the `wpcom_public_coming_soon` option. 

I'm on Create gardening rotation next week so I'll prioritize it and try to debug. Thanks for your patience!

**Comment by ivan-ottinger** [comment]
Another report came today: 27744370-hc

The symptoms looked exactly the same as in one of the comments above: https://github.com/Automattic/wp-calypso/issues/48316#issuecomment-768867543

Switching the site Private and then Public resolved the issue for the user.

**Comment by ramonjd** [comment]
Thanks @ivan-ottinger I've been looking at this. I suspect that it's because we're erroneously [switching off Coming Soon mode](https://github.com/Automattic/wp-calypso/blob/trunk/apps/editing-toolkit/editing-toolkit-plugin/coming-soon/coming-soon.php#L108) due to a race condition.

I have no evidence for this yet as it's not reproducible on a sandbox and only _sometimes_ happens on prod.

I'll keep testing this week and update here.

**Comment by ramonjd** [comment]
So I think we're setting the coming soon option too early. Subsequent changes to the blog's public status trigger a hook that turns coming soon off.

~D57169-code should take care of this while we think of a better solution.~

We thought of a better solution 😆 

D57320-code

The problem only affects certain designs, e.g., Balan. And it started after October 2020.

TL;DR Some of our headstart annotations have explicitly set `wpcom_public_coming_soon` to `0`

**Comment by ramonjd** [comment]
This has been addressed by D57320-code

The reason why it wasn't happening all the time because it was only affecting a few designs, e.g., Balan, Pollard and Gilbert

I'll close for now. Thanks everyone for your help!

Please feel free to reopen if it happens again.

**Comment by noteeth** [comment]
Re-opening this issue after another instance popped up here 3868287-zd
Site migrated in with `full-site-editing` disabled and was stuck at "coming soon" despite being launched.  Public was not an available option until enabling `full-site-editing`.

**Comment by simison** [comment]
@vindl can you help triage this? Thanks! We probably just need [the same changes as were done for some designs](https://github.com/Automattic/wp-calypso/issues/48316#issuecomment-781846373) done also for FSE designs.

Feel free to open new issue and close this, or re-use this if you prefer.

**Comment by ramonjd** [comment]
Are we talking about the TwentyTwentyOne theme?

I've thrown up a patch here: D59544-code

**Comment by vindl** [comment]
> @vindl can you help triage this? Thanks! We probably just need the same changes as were done for some designs done also for FSE designs.

I suspect this is not related to FSE, but it's referring to ETK plugin under its old name.

> Site migrated in with full-site-editing disabled and was stuck at "coming soon" despite being launched. Public was not an available option until enabling full-site-editing.

@noteeth could you clarify if `full-site-editing` here refers to:
- plugin that has `full-site-editing` slug (now called Editing Toolkit Plugin)
- Legacy Dotcom FSE sticker
- Core FSE (under development, shouldn't be live yet)

**Comment by katiebethbrown** [comment]
I ran into this on a user site earlier, but I can't find the interaction so I was testing and I can replicate on [a test site](https://testlaunch960883515.wpcomstaging.com/). 

To replicate:
- AT an unlaunched site
- then deactivate the FSE/WET plugin
- Check the site and see it becomes public even though it isn't launched
- Check the settings, and it shows as unlaunched, with the message about being hidden, but the settings behind the overlay show it as public.

Settings:
<img width="755" alt="Screen Shot 2021-03-31 at 8 49 54 PM" src="https://user-images.githubusercontent.com/42279181/113236734-b1100700-9262-11eb-90eb-158a6203dcba.png">

**Comment by inaikem** [comment]
In direct relation to @katiebethbrown's testing above: 28755935-hc

**Comment by noteeth** [comment]
> @noteeth could you clarify if `full-site-editing` here refers to:
> 
> * plugin that has `full-site-editing` slug (now called Editing Toolkit Plugin)
> * Legacy Dotcom FSE sticker
> * Core FSE (under development, shouldn't be live yet)

@vindl 
In my initial comment I was referring to plugin that has `full-site-editing` slug (now called Editing Toolkit Plugin).  Thanks for the information so I know how to clarify in the future

**Comment by nickpagz** [comment]
I'm not sure if this needs a separate issue reported or not, but logging here for now. 
Similar to the [second comment](https://github.com/Automattic/wp-calypso/issues/48316#issuecomment-756320938) above, I helped a user "Launch" their site while in chat. The panel in Calypso said it was launched, ie Public, but it was still showing the Coming Soon banner on the site.
Interaction: #29114135-hc

**Comment by roo2** [comment]
@nickpagz I tried to reproduce that on a new AT site with a custom domain name but the site launched properly first time, do you remember if you launched the site yourself or if you let the user launch it? ( I'm trying to work out if the old wp-admin UI might have been used to set the `blog_public` status to public without launching the site )
It looks to me from the transcript and audit log like they did use the calypso settings screen (`/settings/general/users-website.blog`) but I remember there was a bug where the old wp-admin screen was being shown (`wp-admin/options-reading.php`.The wp-admin screen doesn't launch AT sites properly, so that's one possible explaination. Otherwise I'm scratching my head 🤔

**Comment by roo2** [comment]
@katiebethbrown I replicated that by removing the FSE plugin, the site becomes launched. 
When we were designing the new coming soon feature we made the choice to implement it in the ETK, and accepted the tradeoff that if the plugin were deactivated, coming soon mode wouldn't work properly. pbAok1-17l-p2#comment-2246
I think that the settings page having that launch button and hiding the settings is unfortunate though, that makes it look like coming soon is still active.
I'll see if I can update the settings page so that if the ETK plugin is deactivated, we no longer see the unlaunched site UI.
